### DISCUSSION TITLE
Seeing your first suggestion - Not working!

### DISCUSSION THREAD

**Discussion by ghost** [community/community#51946] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi,

I just tried to get CoPilot working according to what's described here: https://docs.github.com/en/copilot/getting-started-with-github-copilot#seeing-your-first-suggestion-1

But it doesn't work as described. See in Screencast: https://screenpal.com/watch/c0fflKVaUI2

My Configuration:

![image](https://user-images.githubusercontent.com/6480047/229795478-93804a2d-0460-40d9-97f4-dafd3aa0ee37.png)

My expectation:
- When I write a comment like `// add a sample http client request` -> copilot should respond with something like that I get in https://chat.openai.com/
- When I start a method with a descriptive name like **CalculateBetweenDates(** it should respond with a suggestion like described in your above mentioned getting started guide.

**Comment by ayim** [comment]
@ericbrunner does this work better in files with more existing code in it?

@jeffreyhunter77 what's our take on this scenario here? Should I do screenshare w/ customer to see what's being passed to the server and such?

**Reply by jeffreyhunter77** [reply, marked as answer]
@ayim I think it's pretty clear what's going on. There are a couple things here.

One is the document version mismatch error. If he hasn't already, @ericbrunner should upgrade to the 1.79 version of the extension (released yesterday). It has a fix specifically for this issue. Those messages will still appear in the log, but the extension will automatically retry with an updated document version when that happens.

The other issue is really a model issue, e.g. strange parameters for `CalculateDaysBetweenTwoDates`. One known issue is that examples in the documentation tend to be ineffective over time if you have turned on blocking of suggestions matching public code in settings. That's just because those examples tend to eventually get shared publicly.

**Reply by ghost** [reply]
Hi @jeffreyhunter77 Thanks for getting back that quick. According to your suggestions I updated the GitHub **CoPilot** Visual Studio 2022 Extension to the latest version `1.79`.

Until yesterday my "Suggestions matching public code" setting was :  **Block**
I changed that setting to : **Allow**

![image](https://user-images.githubusercontent.com/6480047/230280913-f3c8c76f-bf4e-4b9e-9394-59d086f22dbd.png)

TBH: Despite the fact that I read all I could get into my hands,, I don't really understand that setting. I guess when setting is **Block**, the **CoPilot** won't give me suggestions from public reachable code bases like open source projects. In contrast, when I **Allow** it , I will get suggestions from public code bases , too. 

But what has that to do with the [getting started](https://docs.github.com/en/copilot/getting-started-with-github-copilot#seeing-your-first-suggestion-1) `int CalculateDaysBetweenTwoDates(` ? 

With my initial attempt the setting was **Block**. So I would expect that the getting started tutorial is **not** based on public code base samples, that get filtered away when I have **Block** as my setting. 

Obviously the getting started sample seems to be based on public code base samples the model is trained on, which get filtered away when setting is **Block**

![image](https://user-images.githubusercontent.com/6480047/230283931-9094661f-db93-4227-8bbb-e03f8b923e8e.png)

**Comment by ghost** [comment]
@ayim Not really. I'll wait until I get access to [Copilot Chat for Visual Studio Preview ](https://devblogs.microsoft.com/visualstudio/github-copilot-chat-for-visual-studio-2022/)

![image](https://user-images.githubusercontent.com/6480047/229989031-0f72d934-a73f-4a55-bcfe-0dd16fafd0e3.png)

![image](https://user-images.githubusercontent.com/6480047/229989251-43efef31-fba8-404d-9b1d-716ebc491b43.png)

![image](https://user-images.githubusercontent.com/6480047/229989676-48c48633-53d5-4567-a614-54bc862c4cb7.png)

![image](https://user-images.githubusercontent.com/6480047/229990627-de860c09-3220-46ff-87c8-016548cd5acb.png)

I added the log from Github CoPilot (Output Window VS2022 Community) here: https://gist.github.com/ericbrunner/545421a24e0f712e44192539b033ca36
